src/index.ts

```diff
@@ -0,0 +1,1368 @@
+                        acc.push(...getParentEventHandler(allInterfacesMap[i]));
+                        return acc;
+                    }, [])
+                    : [];
+
+            return extendedParentWithEventHandler.concat(implementedParentsWithEventHandler);
+        }
+
+        const result: Record<string, Browser.Interface[]> = {};
+        for (const i of allInterfaces) {
+            result[i.name] = getParentsWithEventHandler(i);
+        }
+        return result;
+    })();
+
+    return flavor === Flavor.ES6Iterators ? emitES6DomIterators() : emit();
+
+    // Used to decide if a member should be emitted given its static property and
+    // the intended scope level.
+    function matchScope(scope: EmitScope, x: Browser.Method) {
+        if (scope === EmitScope.All) return true;
+        else if (x.static) return scope === EmitScope.StaticOnly;
+        else return scope === EmitScope.InstanceOnly;
+    }
+
+    /// Parameter cannot be named "default" in JavaScript/Typescript so we need to rename it.
+    function adjustParamName(name: string) {
+        switch (name) {
+            case "default": return "_default";
+            case "delete": return "_delete";
+            case "continue": return "_continue";
+            default: return name;
+        }
+    }
+
+    function getElements<K extends string, T>(a: Record<K, Record<string, T>> | undefined, k: K): T[] {
+        return a ? mapToArray(a[k]) : [];
+    }
+
+    function tryGetMatchingEventType(eName: string, i: Browser.Interface) {
+        if (i.events) {
+            const event = i.events.event.find(e => e.name === eName);
+            return event && event.type;
+        }
+        return undefined;
+    }
+
+    function getEventTypeInInterface(eName: string, i: Browser.Interface) {
+        if (eName === "abort" && (i.name === "IDBDatabase" || i.name === "IDBTransaction" || i.name === "MSBaseReader" || i.name === "XMLHttpRequestEventTarget")) return "Event";
+        else if (eName === "readystatechange" && i.name === "XMLHttpRequest") return "Event";
+        else if (i.name === "XMLHttpRequest") return "ProgressEvent";
+        else {
+            const ownEventType = tryGetMatchingEventType(eName, i);
+            return ownEventType || eNameToEType[eName] || "Event";
+        }
+    }
+
+    /// Determine if interface1 depends on interface2
+    function dependsOn(i1Name: string, i2Name: string) {
+        return iNameToIDependList[i2Name] && iNameToIDependList[i1Name]
+            ? iNameToIDependList[i1Name].indexOf(i2Name) > -1
+            : i2Name === "Object";
+    }
+
+    // Some params have the type of "(DOMString or DOMString [] or Number)"
+    // we need to transform it into [“DOMString", "DOMString []", "Number"]
+    function decomposeTypes(t: string) {
+        return t.replace(/[\(\)]/g, "").split(" or ");
+    }
+
+    function createTextWriter(newLine: string) {
```

Comment:
Might be better to print to nodes eventually?

Reply:
do not think so. so far our transformations are not that complicated..